pyocd/coresight/cortex_m.py

```diff
@@ -1105,3 +1105,10 @@ def exception_number_to_name(self, exc_num, name_thread=False):
                 return "Interrupt[%s]" % name
             else:
                 return "Interrupt %d" % irq_num
+
+    def in_thread_mode_on_main_stack(self):
+        try:
+            return self._target_context.read_core_register('ipsr') == 0 and (self._target_context.read_core_register('control') & 2) == 0
+        except exceptions.TransferError as exc:
+            logging.debug("Transfer error checking mode: %s", exc)
+            return False
```

Comment:
I'm not sure returning False on a transfer failure is the right thing to do. It's probably better to just let the exception pass further up. Returning False says (obviously) that the core is not in Thread mode on MSP, but we really don't know that.